### PR TITLE
Feature: Added support for multiselect list prompt

### PR DESCRIPTION
**Pull request by okeefem2**
@clydin Here is the PR for just the core changes to @angular/cli and @angular-devkit-core. Including the changes you requested on https://github.com/angular/angular-cli/pull/13004. Please let me know if any further changes are necessary! 

I will open a second PR with the guard schematic changes if/when this one is merged in.

Thank you, 

Michael

### CODE REVIEW

File: packages/angular_devkit/core/src/json/schema/interface.ts [hunk 1]
```diff
@@ -99,13 +99,14 @@ export interface PromptDefinition {
   id: string;
   type: string;
   message: string;
-  default?: string | number | boolean | null;
+  default?: string | string [] | number | boolean | null;
```
Comment (clydin) [inline packages/angular_devkit/core/src/json/schema/interface.ts]: extra space on `string[]`

File: packages/angular/cli/models/schematic-command.ts [hunk 2]
```diff
@@ -309,7 +310,8 @@ export abstract class SchematicCommand<
               question.type = 'confirm';
               break;
             case 'list':
-              question.type = 'list';
+              const multiselect = !!definition.raw && (definition.raw as JsonObject)['multiselect'];
+              question.type = !!multiselect ? 'checkbox' : 'list';
```
Comment (clydin) [inline packages/angular/cli/models/schematic-command.ts]: should be able to use `definition.multiselect` directly here.

File: packages/angular_devkit/core/src/json/schema/registry.ts [hunk 3]
```diff
@@ -624,23 +625,44 @@ export class CoreSchemaRegistry implements SchemaRegistry {
...
       map(answers => {
-        for (const path in answers) {
-          const pathFragments = path.split('/').map(pf => {
-            if (/^\d+$/.test(pf)) {
-              return pf;
-            } else {
-              return '\'' + pf + '\'';
-            }
-          });
+        const pathFragmenter = (pf: string) => {
```
Comment (clydin) [inline packages/angular_devkit/core/src/json/schema/registry.ts]: I don't think these changes are needed.  `answers` should always be an object.

File: packages/angular/cli/models/schematic-command.ts (hunk 2 above)
Comment (okeefem2) [reply packages/angular/cli/models/schematic-command.ts]: I was having trouble with this. At this point in the process, definition.multiselect is not present. It is only present in definition.raw. I am guessing there is some step where that is either not being set to the definition or is being lost. Would you have any insight as to where that might be?

File: packages/angular_devkit/core/src/json/schema/registry.ts (hunk 3 above)
Comment (okeefem2) [reply packages/angular_devkit/core/src/json/schema/registry.ts]: You are right! I guess when I was first working on this the answers were coming through as an array of the selected values. I realize now that was an artifact of my erroneous test case. I will make the change right away! For making changes, what is the standard for the commits? @clydin

Comment (clydin) [reply packages/angular_devkit/core/src/json/schema/registry.ts]: The process is generally to make additional commits locally; then squash (`git rebase -i master`) the fix-up commits into the relevant previously existing commit.

File: packages/angular/cli/models/schematic-command.ts (hunk 2 above)
Comment (okeefem2) [reply packages/angular/cli/models/schematic-command.ts]: I made the requested changes minus this one... When testing this I am still having the issue where multiselect is not present at this point. I can see that it is present at some point before this, but  it is being stripped out at some point but I am having trouble identifying where.

Comment (okeefem2) [reply packages/angular/cli/models/schematic-command.ts]: I think I have possibly figured out what is wrong. The built version of @angular/cli has a dependency on @angular-devkit/core, looking into the registry.js in the node_modules of @angular/cli I do not see the changes I have made, but looking in dist folder I do see the changes. So if I link the built @angular-devkit/core to the built @angular/cli this should show the changes I expect. Is there a better way to build the packages locally to test changes across multiple packages @clydin?

Comment (okeefem2) [reply packages/angular/cli/models/schematic-command.ts]: Either way, this did work. So all changes have been addressed!

**Pull request merged**